code/neuron_model.py

```diff
@@ -0,0 +1,307 @@
+        return np.tile(tmp, repeats)
+
+    def rhs(self, t, X, Iapp):
+        """
+        Right-hand side of Wang [1] model.
+
+        |  :param X: States of the system
+        |  :param t: Time
+        |  :param Iapp: Injected current (stimulation)
+        |  :return: Calculate membrane potential & activation variables
+        """
+        V, h, H, n = X[0], X[1], X[2], X[3]
+
+        dV = ((-self.I_T(V, h) - self.I_H(V, H) - self.I_Na(V, n) -
+              self.I_K(V, n) - self.I_NaP(V) - self.I_L(V) +
+              Iapp)/self.p['Cm'])
+        dh = (self.p['phi_h']*(self.h_inf(V, self.p['theta_h'], self.p['k_h'])
+              - h)/self.tau_h(V, self.p['theta_h'], self.p['k_h']))
+        dH = self.p['phi_H'] * (self.H_inf(V) - H) / self.tau_H(V)
+        dn = (self.p['phi_K'] * (self.n_inf(V, self.p['sigma_K']) - n) /
+              self.tau_n(V, self.p['sigma_K']))
+
+        return np.array([dV, dh, dH, dn])
+
+    def run(self, tf, vname, tname, int_name='vode', *args):
+        """ 
+        Run the model by integrating equations.
+
+        |  :param tf: Total time (ms)
+        |  :param vname: Filename for storing membrane potential
+        |  :param tname: Filename for storing time
+        |  :param int_name: Integration method
+        |  :param *args: If int_method is vode, zvode, lsoda user can choose 
+                         Adams or BDF methods
+        """
+        t0 = 0.0         # Initial time (ms)
+        dt = self.dt     # Integration time step (ms)
+
+        # Some temporal conversions
+        t_sim = int(tf / dt)        # Convert tf (ms) to simulation time
+        P0_sim = int(self.P0 / dt)  # Convert period (ms) to simulation period
+        p_sim = int(self.P / dt)    # Convert duration (ms) to sim. duration
+
+        # Define the input current
+        if self.inp_type == "const":
+            Iapp = np.ones((t_sim, )) * self.amplitude
+        else:
+            Iapp = self.I_ext(P0_sim, p_sim, t_sim)
+            if t_sim > Iapp.shape[0]:
+                raise ValueError('Simulation time exceeds input signal dimensions!')
+
+        V0 = self.p['V0']
+        h0 = self.h_inf(V0, self.p['theta_h'], self.p['k_h'])
+        H0 = self.H_inf(V0)
+        n0 = self.n_inf(V0, self.p['sigma_K'])
+        x0 = [V0, h0, H0, n0]
+
+        # Set parameters and initialize the solver
+        if len(args) == 0:
+            r = ode(self.rhs).set_integrator(int_name)
+        else:
+            r = ode(self.rhs).set_integrator(int_name, method=args[0])
+        r.set_initial_value(x0, t0).set_f_params(Iapp[0])
+
+        v, t = [], []
+        i = 0
+        while r.successful() and r.t < (tf-1):
+            r.set_f_params(Iapp[i])
+            v.append(r.integrate(r.t+dt))
+            t.append(r.t)
+            i += 1
+
+        v = np.array(v)
+        t = np.array(t)
+
+        # np.save("../data/"+vname+"stimulus", Iapp)
+        # np.save("../data/"+vname, v)
+        # np.save("../data/"+tname, t)
```

Comment:
Why is the output code commented out, while the docstring suggests that the method will write results to file?